src/solvers/lowering/byte_operators.cpp

```diff
@@ -1034,6 +1025,9 @@ exprt lower_byte_extract(const byte_extract_exprt &src, const namespacet &ns)
   byte_extract_exprt unpacked(src);
   unpacked.op() = unpack_rec(
     src.op(), little_endian, lower_bound_int_opt, upper_bound_int_opt, ns);
+  CHECK_RETURN(
+    to_bitvector_type(to_type_with_subtype(unpacked.op().type()).subtype())
+      .get_width() == 8);
```

Comment:
Good call, should stop a similar issue cropping up going forward a bit earlier.

That said, I'm a bit worried about all the hard-coded 8s and 7s in this module

Reply:
Ack, I really should revive #917, which would sort this out.